app/dao/templates_dao.py

```diff
@@ -109,8 +109,7 @@ def dao_get_template_by_id(
                 return TemplateHistory.from_json(template_cache_decoded), template_cache_decoded
             else:
                 return Template.from_json(template_cache_decoded), template_cache_decoded
-
-    elif version is not None:
+    if version is not None:
```

Comment:
this should be an elif as if the template isn't returned from cache, it might still be in TemplateHistory

Reply:
good catch!